indica/workflow/observers.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+
+from typing import Any
+from typing import Callable
+from typing import List
+from typing import Optional
+from typing import Protocol
+
+
+class TypeObservable(Protocol):
+    """
+    Typing for Observable class
+    """
+
+    data: Any
+
+    def add_observer(self, observer):
+        pass
+
+    def remove_observer(self, observer):
+        pass
+
+    def clear_observers(self):
+        pass
+
+    def notify_observers(self):
+        pass
+
+
+class TypeObserver(TypeObservable):
+    """
+    Typing for Observer class
+    """
+
+    data: Any
+
+    def update(self, *args, **kwargs):
+        pass
+
+
+class Observable:
+    """
+    Class to hold some data, with information about Observer classes, and
+    signal observers to update when stored data has changed
+    """
+
+    def __init__(self, initial_value: Optional[Any] = None):
+        self.observers: List[TypeObserver] = []
```

Comment:
I think instead of `TypeObserver` and `TypeObservable` we can put the types in quotes like this:

```suggestion
        self.observers: List['Observer'] = []
```